indico/web/client/js/react/components/principals/Search.jsx

```diff
@@ -221,30 +221,86 @@ const searchFactory = config => {
     withEventPersons,
     eventId,
   }) => {
-    const [result, setResult] = useState(null);
+    const [loading, setLoading] = useState(false);
+    const [result, _setResult] = useState(null);
+    const lastResult = useRef(null);
+    const setResult = value => {
+      lastResult.current = result;
+      _setResult(value);
+    };
```

Comment:
just wondering.. do we need a ref for this? or can we use useReducer for result+lastResult where sending a new result sets the last result to the previous one?

Reply:
Yes, I don't see why not, `useRef` just seems simpler as it won't have any side-effects (we just want to silently keep track of the results), but since both are being assigned at the exact same time it's pretty much the same (without a reducing logic):
```
  lastResult.current = result;
  _setResult(value)
```
I can do both.

Reply:
yeah nothing wrong with keeping it as it is right now :)